lib/collection.js

```diff
@@ -1559,14 +1858,33 @@ Collection.prototype.stats = function(options, callback) {
  * @param {Collection~findAndModifyCallback} [callback] The collection result callback
  * @return {Promise<Collection~findAndModifyWriteOpResultObject>} returns Promise if no callback passed
  */
+const findOneAndDeleteSchema = {
+  collation: { type: 'object' },
+  projection: { type: 'object' },
+  sort: { type: 'object' },
+  maxTimeMS: { type: 'number' },
+  session: { type: ClientSession },
+  fields: { overrideOnly: true },
```

Comment:
make a ticket for `fields` to be internally renamed to `projection` (or something).